src/mem.c

```diff
@@ -18,3 +25,72 @@ void avifFree(void * p)
 {
     free(p);
 }
+
+#else
+
+// This implementation rounds up all memory allocations to the nearest 4k (page size), allocates
+// using VirtualAlloc(), and then records the allocation in a linked list. When avifFree is called,
+// instead of freeing the memory, it simply revokes all read and write access to that region
+// forever. If any use-after-free issues are discovered (via fuzzing or the like), enabling this
+// should immediately catch it in a debugger right when it happens.
+
+#include <stdio.h>
+#include <windows.h>
+
+typedef struct Allocation
+{
+    void * ptr;
+    size_t originalSize;
+    size_t size;
```

Comment:
Optional: If I understand the MSDN pages for VirtualAlloc() and VirtualProtect() correctly, we can just pass the original size to these two functions and they will round the size up to the next page boundary for us.